Makefile

```diff
@@ -1,5 +1,20 @@
 .DEFAULT_GOAL := default
 
+include config.mk
+
+SERVICES := auth batch ci memory notebook monitoring website
+SERVICES_IMAGES := $(patsubst %, %-image, $(SERVICES))
+SERVICES_MODULES := $(SERVICES) gear web_common
+CHECK_SERVICES_MODULES := $(patsubst %, check-%, $(SERVICES_MODULES))
+
+HAILTOP_VERSION := hail/python/hailtop/hail_version
+SERVICES_IMAGE_DEPS = hail-ubuntu-image $(HAILTOP_VERSION) $(shell git ls-files hail/python/hailtop gear web_common)
+
+EMPTY :=
+SPACE := $(EMPTY) $(EMPTY)
+EXTRA_PYTHONPATH := hail/python:$(subst $(SPACE),:,$(SERVICES_MODULES))
+PYTHON := PYTHONPATH=$${PYTHONPATH:+$${PYTHONPATH}:}$(EXTRA_PYTHONPATH) python3
```

Comment:
you're delaying the evaluation of PYTHONPATH until bash-evaluation-time, I don't think that's necessary? Admittedly I don't know immediately how to replicate bash `${X:+$X:}` in make.